src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -0,0 +1,325 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(typeof(BinaryPredictionTransformer<IPredictorProducing<float>>), typeof(BinaryPredictionTransformer), null, typeof(SignatureLoadModel),
+    "", BinaryPredictionTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(MulticlassPredictionTransformer<IPredictorProducing<VBuffer<float>>>), typeof(MulticlassPredictionTransformer), null, typeof(SignatureLoadModel),
+    "", MulticlassPredictionTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(RegressionPredictionTransformer<IPredictorProducing<float>>), typeof(RegressionPredictionTransformer), null, typeof(SignatureLoadModel),
+    "", RegressionPredictionTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public abstract class PredictionTransformerBase<TModel> : IPredictionTransformer<TModel>, ICanSaveModel
+        where TModel : class, IPredictor
+    {
+        protected readonly IHost Host;
+        protected readonly ISchemaBindableMapper BindableMapper;
+        protected readonly ISchema TrainSchema;
+
+        public string FeatureColumn { get; }
+
+        public ColumnType FeatureColumnType { get; }
+
+        public TModel Model { get; }
+
+        public PredictionTransformerBase(IHost host, TModel model, ISchema trainSchema, string featureColumn)
+        {
+            Contracts.CheckValue(host, nameof(host));
+            Host = host;
+            Host.CheckValue(trainSchema, nameof(trainSchema));
+
+            Model = model;
+            FeatureColumn = featureColumn;
+            trainSchema.TryGetColumnIndex(featureColumn, out int col);
+            if (col < 0)
+                throw Host.ExceptSchemaMismatch(nameof(featureColumn), "feature", featureColumn);
+            FeatureColumnType = trainSchema.GetColumnType(col);
+
+            TrainSchema = trainSchema;
+            BindableMapper = ScoreUtils.GetSchemaBindableMapper(Host, model);
+        }
+
+        internal PredictionTransformerBase(IHost host, ModelLoadContext ctx)
+        {
+            Host = host;
+
+            ctx.LoadModel<TModel, SignatureLoadModel>(host, out TModel model, "Model");
```

Comment:
>"Model" [](start = 78, length = 7)

can we make it private const string PredictionDir = "Model";
and use it here and in line 112